pkg/kubectl/resource/builder.go

```diff
@@ -240,6 +240,13 @@ func (b *Builder) RequireNamespace() *Builder {
 	return b
 }
 
+// NameParam accepts the name for a resource. Meant to be used in combination with a default
+// resource type set by ResourceTypes.
+func (b *Builder) NameParam(name string) *Builder {
+	b.names = append(b.names, name)
```

Comment:
`b.names` holds just the names of resources. Types are in `b.resources`. There is a [method](https://github.com/GoogleCloudPlatform/kubernetes/blob/b33954c45e41ddde99dd3e3de358738a04489bd4/pkg/kubectl/resource/builder.go#L180-L185) for adding just types, but there is none for names.